src/DGtal/io/readers/ITKReader.ih

```diff
@@ -188,8 +188,29 @@ namespace DGtal {
       throw IOException();
     }
     
-    const TypeDGtalImage image( itk_image );
-
+    TypeDGtalImage image( itk_image );
+    if (shiftDomainUsingOrigin)
+    {
+      const auto c = itk_image->GetOrigin();
+      typename Domain::Point p;
+      for (unsigned int k = 0; k < dimension; k++)
+      {
+        p[k] = static_cast<typename Domain::Integer>(c[k]);
+      }
+      image.updateDomain(p);
+      //updating only the domain is not sufficient to import correctly
```

Comment:
I would like to reproduce it. Is this exercised in the test?
I mean, if I remove this code, does the test fail?

Maybe we can fix it in the `operator()` of the ImageContainerByITKImage.

Reply:
yes I just complete the test in last commit, you can try to comment these lines and try ./tests/io/readers/testITKReader it will fails with the range() access to image value.

Reply:
If you have the DGtalTools, you can also try with and without these lines and you will have:
<img width="912" alt="Capture d’écran 2020-05-22 à 21 13 04" src="https://user-images.githubusercontent.com/772865/82701772-12457200-9c71-11ea-8872-3620a878be49.png">
And if you comment the lines you will have a shifted display:
<img width="912" alt="Capture d’écran 2020-05-22 à 21 12 40" src="https://user-images.githubusercontent.com/772865/82701933-56387700-9c71-11ea-94bb-f45dc49ab38d.png">

Reply:
> Maybe we can fix it in the operator() of the ImageContainerByITKImage.
I try to do some edits on the iterator but it brakes your tests.